storage/src/main/java/org/apache/kafka/storage/internals/log/SnapshotFile.java

```diff
@@ -63,7 +63,7 @@ public File file() {
     public void renameTo(String newSuffix) throws IOException {
         File renamed = new File(Utils.replaceSuffix(file.getPath(), "", newSuffix));
         try {
-            Utils.atomicMoveWithFallback(file.toPath(), renamed.toPath());
+            Utils.atomicMoveWithFallback(file.toPath(), renamed.toPath(), false);
```

Comment:
This works since it's ok to lose a file to be deleted. Perhaps it's better to rename the method to sth like `renameToDelete` so that it's clear that this is not a generic method for arbitrary renaming.